src/libsyntax/errors/mod.rs

```diff
@@ -312,6 +312,10 @@ impl<'a> DiagnosticBuilder<'a> {
         self
     }
 
+    pub fn span(&self) -> &MultiSpan {
+        &self.span
+    }
+
```

Comment:
I think a pub field would be nicer (more idiomatic Rust), could change message too I guess